pkg/types/common/common.go

```diff
@@ -0,0 +1,85 @@
+package common
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+
+	"github.com/go-playground/validator/v10"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// FencingCredential stores the information about a baremetal host's management controller.
+type FencingCredential struct {
```

Comment:
Will baremetal platform be using this struct for fencing instead of existing BMC? If not, then maybe it should be none platform only.

Reply:
Yes, we plan to use it in baremetal as well.
However it'll be mutually exclusive with the bmc credentials for the baremetal use case.